src/block-management/block-holder.js

```diff
@@ -197,6 +198,28 @@ export class BlockHolder extends React.Component<PropsType, StateType> {
 		);
 	}
 
+	getAccessibilityLabel() {
+		const { title, attributes, name } = this.props;
+
+		if ( name === getUnregisteredTypeHandlerName() ) { // is the block unrecognized?
+			return title; //already localized
+		}
+
+		const blockName = sprintf(
+			/* translators: accessibility text. %s: block name. */
+			_x( '%1$s block' ),
+			title, //already localized
+		);
+
+		const blockType = getBlockType( name );
+		if ( blockType.getAccessibilityLabel ) {
+			const blockAccessibilityLabel = blockType.getAccessibilityLabel( attributes ) || '';
+			return blockName + ( blockAccessibilityLabel ? '. ' + blockAccessibilityLabel : '' );
```

Comment:
I am aiming towards to the current approach because it brings consistency. With this, the description for all blocks starts with 'X block. '

on the other hand I think the '. ' and concatenations needs a revisit as we discussed earlier for localization issues. would it be better to do this as follows?

```
             sprintf(
			/* translators: accessibility text. 1: block name. 2: block content information. */
			_x( '%1$s block. %2$s', 'Accessibility text for a block' ),
			blockName,
			blockAccessibilityLabel
		);
```